app/controllers/admin/cases_controller.rb

```diff
@@ -30,55 +34,118 @@ def index
   end
 
   def new
-    case_creator = CTS::Cases::Create.new(Rails.logger, case_model: Case::Base)
+    if params[:correspondence_type].present?
+      @correspondence_type = params[:correspondence_type]
+      self.send("prepare_new_#{@correspondence_type}".to_sym)
```

Comment:
Not worth changing now, but just wanted to mention that I think that the `to_sym` here may be superfluous, so would be more succinct to just not bother. Also, it's possible that Ruby internally converts the sym back to a string, I don't know how `send` works.